# Genomic insights on the evolution of chemosensory receptor genes across different tick species

**Chemosensation is essential for ticks, helping them locate habitats, hosts, mates, and oviposition sites. Chemosensory receptor genes â€” including olfactory and gustatory receptor genes â€” play a central role in this process. Different tick species occupy diverse ecological niches and vary in life cycles and host preferences (e.g., reptiles, mammals, birds). As a result, they may have evolved distinct repertoires of chemosensory receptor genes to adapt to their specific environments and lifestyles.**

This repository contains all the data, scripts, programs, and step-by-step instructions for performing a comparative genomic analysis across 14 tick species, with a special focus on the castor bean tick, ***Ixodes ricinus***. The goal is to explore patterns in chemosensory gene evolution and provide a reproducible pipeline for others working on similar questions in tick genomics.

## **Programs and Tools**

**If you are working on an HPC system from your institution, the programs you need are likely already installed. In my case, I used the **[UPPMAX](https://www.uu.se/en/centre/uppmax) from Uppsala University, [Dardel HPC system](https://www.pdc.kth.se/hpc-services/computing-systems/dardel-hpc-system) from KTH and [COSMOS](https://www.lunarc.lu.se/systems/cosmos/) from Lund University â€” itâ€™s always better to have access to more computing and storage resources, especially if you're trying to save your PhD (like I was!).

**Personally, I prefer to install software using **[conda](https://anaconda.org/anaconda/conda), [mamba](https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html), or [Singularity](https://docs.sylabs.io/guides/3.0/user-guide/installation.html). Many people rely on their HPC support teams to install software, which can sometimes take a while. So, if you have the skills (or the curiosity) to do it yourself and save time, go for it! Of course, if your HPC support team is fast and efficient, that's also a great option.

`QUAST (v5.3.0)`

`BUSCO (6.0.0)`

`ncbi-datasets-cli`

`orthofinder ()`

`CAFE ()`

## **Directory Structure**

```
tick_genomics/
â”œâ”€â”€ data/
â”‚ Â  â”œâ”€â”€ genomes/
â”‚ Â  â”‚ Â  â”œâ”€â”€ A_americanum/
â”‚ Â  â”‚ Â  â”œâ”€â”€ A_maculatum/
â”‚ Â  â”‚ Â  â”œâ”€â”€ D_albipictus/
â”‚ Â  â”‚ Â  â”œâ”€â”€ D_andersoni/
â”‚ Â  â”‚ Â  â”œâ”€â”€ D_silvarum/
â”‚ Â  â”‚ Â  â”œâ”€â”€ D_variabilis/
â”‚ Â  â”‚ Â  â”œâ”€â”€ H_asiaticum/
â”‚ Â  â”‚ Â  â”œâ”€â”€ H_longicornis/
â”‚ Â  â”‚ Â  â”œâ”€â”€ I_persulcatus/
â”‚ Â  â”‚ Â  â”œâ”€â”€ I_ricinus/
â”‚ Â  â”‚ Â  â”œâ”€â”€ I_scapularis/
â”‚ Â  â”‚ Â  â”œâ”€â”€ O_turicata/
â”‚ Â  â”‚ Â  â”œâ”€â”€ R_microplus/
â”‚ Â  â”‚ Â  â”œâ”€â”€ R_sanguineus/
â”‚ Â  â”‚ Â  â”œâ”€â”€ logs/
â”‚ Â  â”‚ Â  â””â”€â”€ tick_species_list.txt
â”‚ Â  â””â”€â”€ receptors/
â”œâ”€â”€ scripts/
â”‚ Â  â””â”€â”€ busco_downloads/
â”‚ Â  Â  Â  â””â”€â”€ lineages/
â”‚ Â  Â  Â  Â  Â  â””â”€â”€ arthropoda_odb10/
â”‚ Â  Â  Â  Â  Â  Â  Â  â”œâ”€â”€ hmms/
â”‚ Â  Â  Â  Â  Â  Â  Â  â”œâ”€â”€ info/
â”‚ Â  Â  Â  Â  Â  Â  Â  â””â”€â”€ prfl/
â”œâ”€â”€ analysis/
â”‚ Â  â”œâ”€â”€ busco/
â”‚ Â  â”‚ Â  â””â”€â”€ [Species-specific BUSCO outputs]
â”‚ Â  â”œâ”€â”€ orthofinder/
â”‚ Â  â””â”€â”€ quast/
â”‚ Â  Â  Â  â””â”€â”€ quast_results/
â”‚ Â  Â  Â  Â  Â  â””â”€â”€ [Species folders with QUAST output]
â”œâ”€â”€ notebooks/ Â  Â  Â  Â  Â # (Optional: for Jupyter or Markdown analysis)
â””â”€â”€ README.md
```

## Data Collection

**For this analysis, I used both ****genomic** and **transcriptomic** data from multiple public repositories, mainly [GenBank](https://www.ncbi.nlm.nih.gov/genbank/) and [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/). Additionally, I generated new **RNA-seq data** as part of this study to complement the available resources and ensure better coverage for *Ixodes ricinus*, the focal species.

**Below youâ€™ll find a table with details of the genomes used, including the assembly level, accession number, and how to download each dataset using the **`datasets` command-line tool from NCBI (if available). This made it easier to keep things organized and reproducible. **I really recommend using this tool to avoid manual errors or inconsistencies.**

**In the case of ***I. ricinus*, I used a curated assembly available through the BIPAA platform because it wasnâ€™t available on NCBI with a formal accession. The download link (use wget) is included in the table as well.

> **ðŸ’¡ ***Tip:* I always keep a note of the day I collected each dataset, it helps to keep track in case versions change later or for reproducibility when writing the methods section.

### Genomic data

| **Specie_name**           | **Assembly_level** | **Accession_number** | **Assembly_name**           | **Publication_date** | **NCBI dataset command or link**                                                                                                                                                                                    | **data_collection_day** |
| ------------------------------- | ------------------------ | -------------------------- | --------------------------------- | -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------- |
| **A. americanum**         | **Scaffold**       | **GCA_030143305.2**  | **ASM3014330v2**            | **May, 2024**        | **datasets download genome accession GCA_030143305.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **A. maculatum**          | **Scaffold**       | **GCA_023969395.1**  | **ASM2396939v1**            | **Jun, 2022**        | **datasets download genome accession GCA_023969395.1 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **D. albipictus**         | **Chromosome**     | **GCA_038994185.2**  | **USDA_Dalb.pri_finalv2**   | **Oct, 2024**        | **datasets download genome accession GCF_038994185.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **22/07/2025**          |
| **D. andersoni**          | **Chromosome**     | **GCA_023375885.3**  | **qqDerAnde1_hic_scaffold** | **Nov, 2024**        | **datasets download genome accession GCF_023375885.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **17/07/2025**          |
| **D. silvarum**           | **Chromosome**     | **GCA_013339745.2**  | **BIME_Dsil_1.4**           | **Jun, 2022**        | **datasets download genome accession GCF_013339745.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **D. variabilis**         | **Contig**         | **GCA_049308735.1**  | **dogtick1.0**              | **Apr, 2025**        | **datasets download genome accession GCA_049308735.1 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **H. asiaticum**          | **Chromosome**     | **GCA_013339685.2**  | **BIME_Hyas_1.3**           | **Jun, 2020**        | **datasets download genome accession GCA_013339685.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **30/07/2025**          |
| **H. longicornis**        | **Chromosome**     | **GCA_013339765.2**  | **BIME_HaeL_1.3**           | **Jun, 2020**        | **datasets download genome accession GCA_013339765.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **I. persulcatus**        | **Chromosome**     | **GCA_964199295.2**  | **IXPE_v2**                 | **Jan, 2017**        | **datasets download genome accession GCA_964199295.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **30/07/2025**          |
| **I. ricinus**            | **Chromosome**     | **:(**               | **:(**                      | **Apr, 2024**        | [https://bipaa.genouest.org/sp/ixodes_ricinus/download/ixodes_ricinus/assembly_1.0/fasta/Iricinus_assembly.fa](https://bipaa.genouest.org/sp/ixodes_ricinus/download/ixodes_ricinus/assembly_1.0/fasta/Iricinus_assembly.fa) | **30/07/2025**          |
| **I. scapularis**         | **Chromosome**     | **GCA_031841145.1**  | **IscapMGN**                | **Sep, 2023**        | **datasets download genome accession GCA_031841145.1 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **16/07/2025**          |
| **Ornithodoros turicata** | **Chromosome**     | **GCA_037126465.1**  | **ASM3712646v1**            | **Mar, 2024**        | **datasets download genome accession GCF_037126465.1  --include  genome,protein,cds,rna,gff3,seq-report   --filename Ornithodoros_turicata_genome.zip**                                                             | **16/07/2025**          |
| **R. microplus**          | **Chromosome**     | **GCA_013339725.1**  | **BIME_Rmic_1.3**           | **Jun, 2020**        | **datasets download genome accession GCF_013339725.1 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **30/07/2025**          |
| **R. sanguineus**         | **Chromosome**     | **GCA_013339695.2**  | **BIME_Rsan_1.4**           | **Jun, 2022**        | **datasets download genome accession GCF_013339695.2 --include  gff3,rna,cds,protein,genome,seq-report**                                                                                                            | **30/07/2025**          |

### Transcriptomic data

## **Genome Quality Assessment**

### QUAST

**I used **[QUAST](https://quast.sourceforge.net/docs/manual.html)  via conda. **QUAST** stands for **Quality ASsessment Tool**, and it's used to evaluate the quality of genome assemblies.

**In simple terms, QUAST helps you ****check how good your genome assemblies are** by calculating useful metrics such as:

* **Total length of the assembly**
* **Number of contigs/scaffolds**
* **N50 and L50 statistics**
* **GC content**
* **Genome completeness (when a reference is provided)**

**It also generates helpful ****interactive visualizations** through a tool called **Icarus**, which makes it easier to explore and compare multiple assemblies.

**For this project, I used ****QUAST v5.3.0** to assess the genome assemblies of all tick species included in the study. This helped me identify differences in assembly quality, which is especially important before running any downstream analyses like gene prediction or orthology detection.

> **ðŸ’¡ ***Tip:* If you're dealing with fragmented or draft assemblies (as is often the case with non-model species), QUAST is a must-have to quickly spot assembly issues or gaps.

**You can install QUAST via **[Conda](https://anaconda.org/bioconda/quast) or check their latest releases and documentation on their [official website](http://quast.sf.net/).

```
#!/bin/bash
#SBATCH -A naiss2024-5-647
#SBATCH -p shared
#SBATCH -c 12 Â  Â  Â  Â  Â  Â  Â  Â  Â  Â  # Number of CPUs for QUAST
#SBATCH --mem=30GB Â  Â  Â  Â  Â  Â  Â   # Total memory for QUAST (adjust if needed)
#SBATCH -t 1-00:00:00
#SBATCH -J quast_combined Â  Â  Â  Â  # Slurm job name
#SBATCH -o logs/quast_combined.out # Standard output file
#SBATCH -e logs/quast_combined.err # Standard error file
#SBATCH --mail-type=ALL
#SBATCH --mail-user=zaide.montes_ortiz@biol.lu.se

# --- Conda Setup (MAKE SURE THESE LINES ARE CORRECT AND UNCOMMENTED) ---
# If your miniconda is in your $HOME and your environment is 'quast_env':
# source $HOME/miniconda3/etc/profile.d/conda.sh
# conda activate quast_env
# or load the module
# -------------------------------------------------------------------------

# Main paths
BASE_DIR=/cfs/klemming/projects/supr/naiss2025-23-132/Ticks/data/genomes
COMBINED_RESULTS_DIR=$BASE_DIR/quast_combined_results Â # Output directory for all combined results

# Create the combined results directory if it doesn't exist
mkdir -p "$COMBINED_RESULTS_DIR"

# Build the list of all assembly FASTA files
# IMPORTANT NOTE: The '-maxdepth 4' is key here.
# It searches:
# Level 0 â†’ $BASE_DIR
# Level 1 â†’ Species folder (e.g., A_americanum)
# Level 2 â†’ 'data' folder
# Level 3 â†’ GCA_/GCF_ folder
# Level 4 â†’ .fna file
ALL_FASTA_FILES=$(find "$BASE_DIR" -maxdepth 4 -type f -name "*.fna" | tr '\n' ' ')

# Check that FASTA files were found before running QUAST
if [ -z "$ALL_FASTA_FILES" ]; then
 Â  Â echo "ERROR: No assembly FASTA files matching *.fna found in the expected structure." >&2
 Â  Â echo "Search path: $BASE_DIR" >&2
 Â  Â echo "Make sure your .fna files are under subdirectories like: $BASE_DIR/Species/data/GCA_ID/file.fna" >&2
 Â  Â exit 1
fi

# Print which FASTA files will be processed (useful for debugging)
echo "The following FASTA files will be processed:"
echo "$ALL_FASTA_FILES"
echo ""

# QUAST flags setup
# Assuming all are eukaryotic genomes
QUAST_FLAGS="--eukaryote"

# If most of your assemblies are fragmented, you can add:
# QUAST_FLAGS="$QUAST_FLAGS --fragmented"

# Run QUAST on all assemblies
# The thread number (-t) must not exceed the CPUs requested (-c) in SBATCH
echo "Starting combined QUAST run..."
quast.py -t 12 -o "$COMBINED_RESULTS_DIR" $QUAST_FLAGS $ALL_FASTA_FILES

echo "Combined QUAST process completed."
echo "Results are saved in: $COMBINED_RESULTS_DIR"
echo "Check 'report.html' and 'transposed_report.tsv' for figures and comparative metrics."
```

### BUSCO

[BUSCO](https://busco.ezlab.org/) stands for **Benchmarking Universal Single-Copy Orthologs**. It's a tool used to **evaluate the completeness of genome assemblies** (or gene annotations) by searching for a set of highly conserved genes expected to be present in nearly all species within a given lineage.

**Think of BUSCO as a way to answer:**

> **"How complete is my genome?"**

### ðŸ” What Does BUSCO Actually Do?

**BUSCO scans your genome (or transcriptome/protein set) for universal single-copy orthologs genes that should ****almost always be present and single-copy** in your species group.

**It gives you results like:**

* **Complete**: The gene was found and looks good
* **Duplicated**: Found more than once (e.g. possible assembly duplication)
* **Fragmented**: Found only partially
* **Missing**: Not found at all

**For this tick genomics project, I used BUSCO with the **`arthropoda_odb10` dataset (insects, ticks, crustaceans, etc.) to check the quality of genome assemblies across all 14 tick species. This is super helpful before doing any gene prediction or downstream analysis â€” you want to be sure your genome isn't missing large chunks!

> **ðŸ’¡ ***Tip:* A BUSCO completeness score above ~90% usually means youâ€™re working with a pretty solid genome.

**You can run BUSCO easily on the command line with:**

```
busco -i your_genome.fna -l arthropoda_odb10 -m genome -o output_name
```

**Here is the script I used:**

```
#!/bin/bash
#SBATCH -A naiss2024-5-647
#SBATCH -p shared
#SBATCH -c 8 Â  Â  Â  Â  Â  Â  Â  Â  Â  Â  Â  Â  # Number of CPUs
#SBATCH --mem=120GB Â  Â  Â  Â  Â  Â  Â  Â  # Total memory
#SBATCH -t 3-00:00:00 Â  Â  Â  Â  Â  Â  Â  # Max wall time: 3 days
#SBATCH -J busco_ticks Â  Â  Â  Â  Â  Â   # Job name
#SBATCH --array=4-13 Â  Â  Â  Â  Â  Â  Â   # Run as an array job for species 4 to 13
#SBATCH --mail-type=ALL Â  Â  Â  Â  Â   # Send email on all job events
#SBATCH --mail-user=zaide.montes_ortiz@biol.lu.se
#SBATCH -o /cfs/klemming/projects/supr/naiss2025-23-132/Ticks/analysis/busco/logs/busco_%A_%a.out
#SBATCH -o /cfs/klemming/projects/supr/naiss2025-23-132/Ticks/analysis/busco/logs/busco_%A_%a.out

# --- Load conda environment (adjust these lines if needed) ---
# source $HOME/miniconda3/etc/profile.d/conda.sh
# conda activate BUSCO
# or load your module via module load ...

# --- Define paths ---
BASE_DIR=/cfs/klemming/projects/supr/naiss2025-23-132/Ticks
GENOME_DIR=$BASE_DIR/data/genomes
OUTPUT_DIR=$BASE_DIR/analysis/busco
SPECIES_LIST=$BASE_DIR/scripts/tick_species_list.txt

# --- Get species name for current SLURM task ---
SPECIES=$(sed -n "$((SLURM_ARRAY_TASK_ID + 1))p" $SPECIES_LIST)

# --- Find the .fna file (genome assembly) ---
# Search recursively inside the 'data' folder for each species
FASTA=$(find "$GENOME_DIR/$SPECIES/data" -type f -name "*_genomic.fna" | head -n 1)

# --- Check if FASTA file was found ---
if [[ ! -f "$FASTA" ]]; then
 Â  Â echo "ERROR: No .fna file found for $SPECIES in $GENOME_DIR/$SPECIES/data/" >&2
 Â  Â exit 1
fi

# --- Create output directory for this species ---
OUT_SPECIES="$OUTPUT_DIR/$SPECIES"
mkdir -p "$OUT_SPECIES"

# --- Run BUSCO ---
# Note: The -o flag is used for the result prefix inside --out_path
busco -i "$FASTA" \
 Â  Â  Â -l arthropoda_odb10 \
 Â  Â  Â -o "$SPECIES" \
 Â  Â  Â -m genome \
 Â  Â  Â -c 8 \
```
